analysis/dataflow/intra_procedural_instruction_ops.go

```diff
@@ -67,10 +67,7 @@ func (state *IntraAnalysisState) DoDebugRef(*ssa.DebugRef) {
 
 // DoUnOp analyzes unary operations and checks the operator to see whether is a load or a channel receive
 func (state *IntraAnalysisState) DoUnOp(x *ssa.UnOp) {
-	switch x.Op {
-	case token.ARROW:
-		state.optionalSyntheticNode(x, x, x)
```

Comment:
Don't we still need the `token.ARROW` case to ensure data-flow that data-flow propagates to the l-value of a channel receive expression in cases where the channel is not a source?

Reply:
The change shouldn't affect the behaviour outside marking the synthetic nodes. On line 74 in that function the `simpleTransfer(state, x, x.X, x)` call is moving the marks from the channel `x.X` to the l-value `x`.